src/v/datalake/translation/partition_translator.cc

```diff
@@ -340,6 +340,32 @@ partition_translator::do_translation_for_range(
     co_return std::move(result.value());
 }
 
+std::optional<size_t> partition_translator::translation_backlog() const {
```

Comment:
I'm curious if there's actually correlation between translation time and bytes on disk, particularly in the face of compressed batches, or complex schemas. I guess the thinking is larger messages likely imply more complex payload, but just thinking out loud I wonder if something as simple as number of records pending would be usable here.

Not really advocating for any changes, but just thought I'd pose the question, and also point out that this won't work for read replicas or restored/FPM-ed partitions without implementing a similar sizing method for cloud.

Reply:
I strongly agree with your point i.e. the actual amount of work is definitely dependent from schema complexity, compression, message size, etc. I think this doesn't really change the relation that the `backlog = some_coefficient * bytes_to_translate` in other words cpu cycles required for translation will always be proportional to number of bytes to translate. I think the parameters you mentioned will influence the proportion coefficient. 
In future we may modify this function to factor in the other complexities.

Reply:
>cpu cycles required for translation will always be proportional to number of bytes to translate

This is what I'm curious about specifically: if two records have the same schema but one has 1 byte per field while the other has 100 bytes per field, does the latter end up taking meaningfully more CPU for parquet writing? I guess size affects things like comparisons, and flush frequency maybe. Just curious if we know if translation is dominated by those factors, vs schema parsing.